fictional dataset related to show business, such as the box office earnings of three different genres of movies (Action, Comedy, and Drama) over a weekend. We'll then perform a one-way ANOVA test to see if there's a significant difference in the earnings among these genres.


In [12]:
import scipy.stats as stats
import numpy as np

# Generate the dataset
action_earnings = [10, 15, 14, 12, 13]
comedy_earnings = [8, 9, 7, 6, 10]
drama_earnings = [6, 7, 5, 4, 6]

# Perform one-way ANOVA
f_statistic, p_value = stats.f_oneway(action_earnings, comedy_earnings, drama_earnings)

# Output the results
f_statistic, p_value


(26.880000000000003, 3.6924672247300125e-05)

The results of the one-way ANOVA test are:
F-statistic: 26.88
P-value: 3.692e-05
Interpretation
The F-statistic value of 26.88 indicates that there is a significant variation between the earnings of the different movie genres. The P-value is very low (3.692e-05), which is much less than the common significance level of 0.05.
This means that we reject the null hypothesis and conclude that there is a statistically significant difference in the box office earnings among the Action, Comedy, and Drama genres. ​es. ​s. ​